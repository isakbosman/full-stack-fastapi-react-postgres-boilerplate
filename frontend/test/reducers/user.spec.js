import reducer from 'reducers/user';
import { ActionTypes } from 'constants/index';

describe('User', () => {
  it('should return the initial state', () => {
    expect(reducer.user(undefined, {})).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.USER_LOGIN}`, () => {
    expect(reducer.user(undefined, { type: ActionTypes.USER_LOGIN })).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.USER_LOGIN_SUCCESS}`, () => {
    expect(reducer.user(undefined, { type: ActionTypes.USER_LOGIN_SUCCESS })).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.USER_LOGOUT}`, () => {
    expect(reducer.user(undefined, { type: ActionTypes.USER_LOGOUT })).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.USER_LOGOUT_SUCCESS}`, () => {
    expect(reducer.user(undefined, { type: ActionTypes.USER_LOGOUT_SUCCESS })).toMatchSnapshot();
  });
});
