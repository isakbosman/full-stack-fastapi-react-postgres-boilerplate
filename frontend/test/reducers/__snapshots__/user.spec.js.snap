// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should handle USER_LOGIN 1`] = `
Object {
  "isAuthenticated": false,
  "status": "running",
}
`;

exports[`User should handle USER_LOGIN_SUCCESS 1`] = `
Object {
  "isAuthenticated": true,
  "status": "ready",
}
`;

exports[`User should handle USER_LOGOUT 1`] = `
Object {
  "isAuthenticated": false,
  "status": "running",
}
`;

exports[`User should handle USER_LOGOUT_SUCCESS 1`] = `
Object {
  "isAuthenticated": false,
  "status": "idle",
}
`;

exports[`User should return the initial state 1`] = `
Object {
  "isAuthenticated": false,
  "status": "idle",
}
`;
